hack/lib/mesh.bash

```diff
@@ -0,0 +1,277 @@
+      policy:
+        autoscaleEnabled: false
+      telemetry:
+        autoscaleEnabled: false
+        resources:
+          requests:
+            cpu: 100m
+            memory: 1G
+          limits:
+            cpu: 500m
+            memory: 4G
+
+    pilot:
+      autoscaleEnabled: false
+    kiali:
+      enabled: true
+    grafana:
+      enabled: true
+    tracing:
+      enabled: true
+    prometheus:
+      enabled: true
+EOF
+
+  logger.info "Waiting until service mesh deployments are available"
+  timeout 600 "[[ \$(oc get deploy -n istio-system ${istio_deployments} --no-headers | wc -l) != 5 ]]" || return 1
+  oc wait --for=condition=Available deployment ${istio_deployments} --timeout=300s -n istio-system || return $?
+}
+
+
+# add_smmr adds ServiceMeshMemberRoll, Networkpolicy and label.
+# It needs to call after deploy knative-serving.
+function add_smmr {
+  cat <<EOF | oc apply -f -
+apiVersion: maistra.io/v1
+kind: ServiceMeshMemberRoll
+metadata:
+  name: default
+  namespace: istio-system
+spec:
+  members:
+    # a list of projects joined into the service mesh
+    - knative-serving
+    - knative-serving-ingress
+    - knative-eventing
+    - default
+EOF
+
+  # Allow webhook to access from kube API.
+  cat <<EOF | oc apply -f -
+apiVersion: networking.k8s.io/v1
+kind: NetworkPolicy
+metadata:
+  name: allow-webhook
+  namespace: knative-serving
+spec:
+  ingress:
+  - {}
+  podSelector:
+    matchLabels:
+      app: webhook
+  policyTypes:
+  - Ingress
+EOF
+
+  cat <<EOF | oc apply -f -
+apiVersion: networking.k8s.io/v1
+kind: NetworkPolicy
+metadata:
+  name: eventing-webhook
+  namespace: knative-eventing
+spec:
+  ingress:
+  - {}
+  podSelector:
+    matchLabels:
+      app: eventing-webhook
+  policyTypes:
+  - Ingress
+EOF
+
+
+  cat <<EOF | oc apply -f -
+apiVersion: networking.k8s.io/v1
+kind: NetworkPolicy
+metadata:
+  name: kourier-gateway
+  namespace: knative-serving-ingress
+spec:
+  ingress:
+  - {}
+  podSelector:
+    matchLabels:
+      app: 3scale-kourier-gateway
+  policyTypes:
+  - Ingress
+EOF
+
+  # serving
+  oc patch deployments.apps -n knative-serving activator -p '{"spec":{"template":{"metadata":{"annotations":{"sidecar.istio.io/inject":"true"}}}}}'
+  oc patch deployments.apps -n knative-serving autoscaler -p '{"spec":{"template":{"metadata":{"annotations":{"sidecar.istio.io/inject":"true"}}}}}'
+  oc patch deployments.apps -n knative-serving autoscaler-hpa -p '{"spec":{"template":{"metadata":{"annotations":{"sidecar.istio.io/inject":"true"}}}}}'
+  oc patch deployments.apps -n knative-serving controller -p '{"spec":{"template":{"metadata":{"annotations":{"sidecar.istio.io/inject":"true"}}}}}'
+  oc patch deployments.apps -n knative-serving kn-cli-downloads -p '{"spec":{"template":{"metadata":{"annotations":{"sidecar.istio.io/inject":"true"}}}}}'
+  oc patch deployments.apps -n knative-serving webhook -p '{"spec":{"template":{"metadata":{"annotations":{"sidecar.istio.io/inject":"true"}}}}}'
```

Comment:
Do we really need all of these in the mesh? IIRC the use-case is to get info on requests etc, tracing and whatnot. Isn't just the activator sufficient for that?

Reply:
Yes, activator, kourier-gateway and kourier-controller(kourier-controller needs it to use gRPC for gateway) is sufficient.
Updated.